pyiron_continuum/schroedinger/schroedinger.py

```diff
@@ -0,0 +1,268 @@
+import numpy as np
+from pyiron_continuum.schroedinger.mesh import RectMesh
+from scipy.sparse.linalg import eigsh, LinearOperator
+from abc import ABC, abstractmethod
+from scipy.constants import physical_constants
+import matplotlib.pyplot as plt
+
+try:
+    import k3d
+    k3d_alarm = ImportAlarm()
+except ImportAlarm:
+    k3d_alarm = ImportAlarm(
+        '3d plotting requires the k3d package which is accessible by `conda install -c conda-forge k3d`. Using k3d in '
+        'a jupyter environment further requires the following shell commands: '
+        '`jupyter nbextension install --py --sys-prefix k3d; jupyter nbextension enable --py --sys-prefix k3d`.'
+    )
+
+
+# TODO: Convert to pyiron units
+HBAR = 1  # EV_TO_U_ANGSTROMSQ_PER_SSQ * physical_constants['Planck constant in eV/Hz'][0] / (2 * np.pi)
+M_EL = 1  # physical_constants['electron mass in u'][0]
+KB = physical_constants['Boltzmann constant in eV/K'][0]
+
+__author__ = "Liam Huber"
+__copyright__ = (
+    "Copyright 2021, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "0.1"
+__maintainer__ = "Liam Huber"
+__email__ = "huber@mpie.de"
+__status__ = "development"
+__date__ = "Sep 8, 2021"
+
+
+class _TISEInput(DataContainer):
+    def __init__(self, init=None, table_name='input', lazy=False):
+        super().__init__(init=init, table_name=table_name, lazy=lazy)
+        self._mesh = None
+        self._potential = None
+        self.n_states = 1
+        self.mass = M_EL
+
+    @property
+    def mesh(self) -> RectMesh:
+        return self._mesh
+
+    @mesh.setter
+    def mesh(self, new_mesh: RectMesh):
+        if not isinstance(new_mesh, RectMesh):
+            raise TypeError(f'Meshes must be of type {RectMesh} but got type {type(new_mesh)}')
+        new_mesh.simplify_1d = False
+        self._mesh = new_mesh
+
+    @property
+    def potential(self):
+        return self._potential
+
+    @potential.setter
+    def potential(self, new_potential):
+        self._potential = new_potential
+
+
+class _TISEOutput(DataContainer):
+    def __init__(self, init=None, table_name='input', lazy=False):
+        super().__init__(init=init, table_name=table_name, lazy=lazy)
+        self.energy = None
+        self.psi = None
+        self._rho = None
+
+    @property
+    def _space_axes(self):
+        return tuple(n for n in range(1, len(self.psi.shape)))
+
+    def _broadcast_vector_to_space(self, vector):
+        return vector.reshape(-1, *(1,)*len(self._space_axes))
+
+    def _normalize_per_state(self, states):
+        return states / self._broadcast_vector_to_space(np.linalg.norm(states, axis=self._space_axes))
+
+    def _weight_states(self, states, weights):
+        return states * self._broadcast_vector_to_space(weights)
+
+    @property
+    def rho(self):
+        if self._rho is None and self.psi is not None:
+            self._rho = self._normalize_per_state(self.psi ** 2)
```

Comment:
Linked with lines 87 and 88, I think this line is incorrect in the case of normalizing a probability density (PD). For a PD, the area under the curve needs to be 1. However, for eigenstates to be normalized, using numpy.linalg.norm for the normalization is correct, since the square of the eigenstate (ie, the PD) would then have an area under the curve equal to 1.

Reply:
Yikes, my bad physics! Yes, you're 100% right. I fixed it easily by allowing state normalization to take the normalization order -- L1 normalization is equivalent to sum normalization for the PD (which are all strictly >=0).